nutanix/resource_nutanix_foundation_image_nodes.go

```diff
@@ -1170,3 +1176,41 @@ func expandBlocks(d *schema.ResourceData) ([]*foundation.Block, error) {
 	}
 	return nil, nil
 }
+
+// This method will look into individual node and cluster creation status and create a collection of errors for errored out processes
+func collectIndividualErrorDiagnostics(progress *foundation.ImageNodesProgressResponse) diag.Diagnostics {
+	// create empty diagnostics
+	var diags diag.Diagnostics
+
+	// append errors for failed node imaging
+	for _, v := range progress.Nodes {
+		if utils.Float64Value(v.TimeElapsed) < float64(AggregatePercentComplete) {
+			message := ""
+			for _, v1 := range v.Messages {
+				message += v1 + ". "
+			}
+			diags = append(diags, diag.Diagnostic{
```

Comment:
Can't we directly use `diag.Errorf()`

Reply:
diag.Errorf() returns diag.Diagnostics, but we need to create a diag.Diagnostic for every node failure and eppend them to create a diag.Diagnostics